src/OpenFeatureSDK/OpenFeatureClient.cs

```diff
@@ -257,9 +257,9 @@ public FeatureClient(FeatureProvider featureProvider, string name, string versio
             catch (FeatureProviderException ex)
             {
                 this._logger.LogError(ex, "Error while evaluating flag {FlagKey}. Error {ErrorType}", flagKey,
-                    ex.ErrorDescription);
-                evaluation = new FlagEvaluationDetails<T>(flagKey, defaultValue, ex.ErrorDescription, Reason.Error,
-                    string.Empty);
+                    ex.ErrorType.GetDescription());
+                evaluation = new FlagEvaluationDetails<T>(flagKey, defaultValue, ex.ErrorType, Reason.Error,
+                    string.Empty, ex.Message);

```

Comment:
Not directly related to your change... but is `null` a better choice than `string.Empty` for the variant?

Reply:
I think that `null` is reasonable, and also what I would have done. Reference and nullable types still seem to be the most idiomatic way to handle optionality in C#.

I am not sure how fast the community is moving this way, but C# has supported additional nullable contexts for a bit. Which allows for something akin to null safety with static analysis.
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-reference-types
(Now enabled by default for dotnet 6 and newer templates.)

At which points these `null` values will probably feel a little better.